tests/gar.test.mjs

```diff
@@ -174,930 +537,371 @@ describe('GatewayRegistry', async () => {
-      // TRANSFER 2K IO to our next stubbed address
-      const quantity = 2_000_000_000;
-      const transferMemory = await transfer({
-        recipient: newStubAddress,
-        quantity: quantity,
+  describe('Cancel-Withdrawal', () => {
+    it('should allow cancelling a delegate withdrawal', async () => {
+      const decreaseStakeTimestamp = STUB_TIMESTAMP + 1000 * 60 * 15; // 15 minutes after stubbedTimestamp
+      const stakeQty = 10000000000;
+      const decreaseQty = stakeQty / 2;
+      const decreaseStakeMsgId = 'decrease-stake-message-id-'.padEnd(43, 'x');
+      const { memory: delegatedStakeMemory } = await delegateStake({
+        delegatorAddress,
+        quantity: stakeQty,
+        gatewayAddress: STUB_ADDRESS,
+        timestamp: STUB_TIMESTAMP,
         memory: sharedMemory,
       });
+      // get the gateway before the delegation and cancellation
+      const gatewayBefore = await getGateway({
+        address: STUB_ADDRESS,
+        memory: delegatedStakeMemory,
+      });
 
-      const delegateStakeResult = await handle(
-        {
-          From: newStubAddress,
-          Owner: newStubAddress,
-          Tags: [
-            { name: 'Action', value: 'Delegate-Stake' },
-            { name: 'Quantity', value: quantity }, // 2K IO
-            { name: 'Address', value: STUB_ADDRESS }, // our gateway address
-          ],
-          Timestamp: STUB_TIMESTAMP + 1,
-        },
-        transferMemory,
-      );
-
-      // assert no error tag
-      const errorTag = delegateStakeResult.Messages?.[0]?.Tags?.find(
-        (tag) => tag.Name === 'Error',
-      );
-      assert.strictEqual(errorTag, undefined);
-
-      // check the gateway record from contract
-      const gateway = await handle(
-        {
-          Tags: [
-            { name: 'Action', value: 'Gateway' },
-            { name: 'Address', value: STUB_ADDRESS },
-          ],
-        },
-        delegateStakeResult.Memory,
-      );
-      const gatewayData = JSON.parse(gateway.Messages[0].Data);
-
-      assert.deepEqual(gatewayData.delegates, {
-        [newStubAddress]: {
-          delegatedStake: quantity,
-          startTimestamp: STUB_TIMESTAMP + 1,
-          vaults: [],
-        },
+      const { memory: decreaseStakeMemory } = await decreaseDelegateStake({
+        memory: delegatedStakeMemory,
+        delegatorAddress,
+        decreaseQty,
+        timestamp: decreaseStakeTimestamp,
+        gatewayAddress: STUB_ADDRESS,
+        messageId: decreaseStakeMsgId,
+      });
+      const { memory: cancelWithdrawalMemory } = await cancelWithdrawal({
+        vaultOwner: delegatorAddress,
+        gatewayAddress: STUB_ADDRESS,
+        vaultId: decreaseStakeMsgId,
+        memory: decreaseStakeMemory,
+        timestamp: decreaseStakeTimestamp,
+      });
+      // get the gateway record
+      const gatewayAfter = await getGateway({
+        address: STUB_ADDRESS,
+        memory: cancelWithdrawalMemory,
```

Comment:
@arielmelendez an example of setting up state to test a specific handler/edge case using the functions above